Use a single method to retrieve title and content from Gutenberg

mkevins opened the pull request:
### Related PRs:

`gutenberg`: https://github.com/WordPress/gutenberg/pull/36072
`gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4182

## Description
This synchronizes the content retrieval mechanism for serializing content from the editor, and includes title and content in a single event. See WordPress/gutenberg#36072 for more details.


## Regression Notes
1. Potential unintended areas of impact
Autosave behavior

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing was conducted as part of a task force to address ongoing issues with the autosave mechanism.

3. What automated tests I added (or what prevented me from doing so)
These changes involve modifications to the mechanism to serialize content over the bridge to address a race condition. Creating the necessary configuration to test integration of our React Native bridge implementations and protocols is out of scope for the autosave task force project, however, this could be very valuable as a future project.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mkevins committed:
Use a single method to retrieve title and content from Gutenberg

This modifies the API used to retrieve both the title and serialized
content from the Gutenberg editor from one event, and includes Gutenberg
changes to synchronize access to the latch used to suspend execution in
the mechanism awaiting response.

mkevins committed:
Update gutenberg-mobile reference

ashiagr committed:
Merge branch 'develop' into fix/synchronize-content-serialization-over-the-bridge

# Conflicts:
#	build.gradle

mkevins committed:
Update gutenberg-mobile reference

mkevins committed:
Merge branch 'develop' into fix/synchronize-content-serialization-over-the-bridge

mkevins committed:
Update gutenberg-mobile reference

mkevins committed:
Merge branch 'develop' into fix/synchronize-content-serialization-over-the-bridge

malinajirka commented on WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:
```diff
@@ -1775,8 +1776,16 @@ private void updateAndSavePostAsyncOnEditorExit(@NonNull final OnPostUpdatedFrom
-            String content = (String) mEditorFragment.getContent(oldContent);
+            String title, content;
+            // To reduce redundant bridge events emitted to the Gutenberg editor, we get title and content at once
+            if (mEditorFragment instanceof GutenbergEditorFragment) {
```
❓ I'm wondering if we need this `if` statement. Both Guteneberg fragment and Aztec fragment provide `getTitleAndContent(..)` method, right?

mkevins replied:
Good point! I refactored the implementation to remove `getTitle` method from the interface (since it is no longer used anywhere), and now that method is private in Aztec. This allows us to simplify this and remove the conditional here.

mkevins committed:
Use renamed latch callback

mkevins committed:
Remove unused getTitle method

mkevins committed:
Make Aztec getTitle private

This is no longer an interface method and can be made private.

mkevins committed:
Update gutenberg-mobile reference

mkevins committed:
Merge branch 'develop' into fix/synchronize-content-serialization-over-the-bridge

mkevins committed:
Add release note

mkevins committed:
Refactor updateFromEditor to remove conditional

Since Aztec and Gutenberg can both use the same interface method, there
is no longer a need to condition on the fragment instance type.

antonis reviewed (approved):
Great work @mkevins 👍 
I tested with https://github.com/WordPress/gutenberg/pull/36072#pullrequestreview-798950447 on a Pixel 5 (Android 12) and everything worked as expected. The code changes also LGTM 🎉

mkevins committed:
Update gutenberg-mobile reference

mkevins committed:
Update gutenberg-mobile reference

mkevins commented:
Thank you for reviewing and testing!
